### ISSUE TITLE
Generic struct names generated by schemars are rejected by openapi-lint

### ISSUE THREAD

**Issue by sunshowers** [oxidecomputer/openapi-lint#14]
Let's say I have a generic struct:

```rust
#[derive(JsonSchema, Serialize)]
struct MySerializedStruct<T> {
    ...
}

#[derive(JsonSchema, Serialize)]
struct MyConcreteType;
```

In this case, for `MySerializedStruct<MyConcreteType>`, the type name that schemars generates is `MySerializedStruct_for_MyConcreteType`. This isn't PascalCase, so it gets rejected by openapi-lint.

There is no way to control how this is generated in schemars, as far as I can tell. Using `schemars(rename)` won't work if there are multiple different concrete types that you'd want to generate.

I've worked around this [in an ugly fashion](https://github.com/oxidecomputer/omicron/pull/2830/files#diff-286aa514de30a5d7d781d7490327df4574c76d7c53e6aed9553c86a2cded2508) for now:

```rust
    let mut errors = openapi_lint::validate(&spec);
    errors.retain(|s| {
        !(s.contains("that is not PascalCase") && s.contains("_for_"))
    });
```

But it seems like openapi-lint should detect and skip these cases.

(As another improvement, it would be nice to have the lint errors be more structured so we wouldn't have to do string scraping.)

**Comment by ahl** [comment]
> There is no way to control how this is generated in schemars, as far as I can tell. Using `schemars(rename)` won't work if there are multiple different concrete types that you'd want to generate.

Check out https://github.com/oxidecomputer/omicron/blob/main/nexus/types/src/external_api/shared.rs#L38 for how to do this.

**Comment by sunshowers** [comment]
Aha, thanks! That works really well.
